core/state_transition.go

```diff
@@ -316,6 +393,7 @@ func (st *StateTransition) preCheck() error {
 			return fmt.Errorf("%w: address %v, nonce: %d", ErrNonceMax,
 				msg.From.Hex(), stNonce)
 		}
+		// TODO(pl): This is removed in celo-blockchain, find out why
```

Comment:
As far as I know, there is no actual reason.
I think that was commented in the Espresso hard fork to avoid some test failing to fix it later, and nobody noticed that we never reverted that comment. And the inclusion of this requires a hard fork, and we always forgot to add it.
This fixes a possible attack the requires a ton of money that maybe for us is not an issue right now, but I'm 100% in to leave it from now.
I think that you can remove this TODO